### PR TITLE
132 meta1 jen5

### PR DESCRIPTION
**Pull request by jenlampton**
another PR for https://github.com/backdrop/backdrop-issues/issues/132

### CODE REVIEW

File: core/modules/simpletest/tests/common.test [hunk 1]
```diff
@@ -2053,30 +2053,30 @@ class CommonBackdropWriteRecordTestCase extends BackdropWebTestCase {
     $person->name = 'John';
     $person->unknown_column = 123;
     $insert_result = backdrop_write_record('test', $person);
-    $this->assertTrue($insert_result == SAVED_NEW, t('Correct value returned when a record is inserted with backdrop_write_record() for a table with a single-field primary key.'));
-    $this->assertTrue(isset($person->id), t('Primary key is set on record created with backdrop_write_record().'));
-    $this->assertIdentical($person->age, 0, t('Age field set to default value.'));
-    $this->assertIdentical($person->job, 'Undefined', t('Job field set to default value.'));
+    $this->assertTrue($insert_result == SAVED_NEW, 'Correct value returned when a record is inserted with drupal_write_record() for a table with a single-field primary key.');
+    $this->assertTrue(isset($person->id), 'Primary key is set on record created with drupal_write_record().');
```
Comment (quicksketch) [inline core/modules/simpletest/tests/common.test]: drupal => backdrop on these two lines.

File: core/modules/simpletest/tests/common.test [hunk 2]
```diff
@@ -2084,65 +2084,65 @@ class CommonBackdropWriteRecordTestCase extends BackdropWebTestCase {
...
+    $this->assertIdentical($result->name, 'Paul', 'Name field set.');
+    $this->assertIdentical($result->age, NULL, 'Age field set.');
 
     // Insert a record - do not specify the value of a column that allows NULL.
     $person = new stdClass();
     $person->name = 'Meredith';
     $insert_result = backdrop_write_record('test_null', $person);
-    $this->assertTrue(isset($person->id), t('Primary key is set on record created with backdrop_write_record().'));
-    $this->assertIdentical($person->age, 0, t('Age field set to default value.'));
+    $this->assertTrue(isset($person->id), 'Primary key is set on record created with drupal_write_record().');
```
Comment (quicksketch) [inline core/modules/simpletest/tests/common.test]: drupal => backdrop

File: core/modules/simpletest/tests/common.test [hunk 3]
```diff
@@ -2233,9 +2233,9 @@ class CommonSimpleTestErrorCollectorTestCase extends BackdropWebTestCase {
    * Assert that a collected error matches what we are expecting.
    */
   function assertError($error, $group, $function, $file, $message = NULL) {
-    $this->assertEqual($error['group'], $group, t("Group was %group", array('%group' => $group)));
-    $this->assertEqual($error['caller']['function'], $function, t("Function was %function", array('%function' => $function)));
-    $this->assertEqual(backdrop_basename($error['caller']['file']), $file, t("File was %file", array('%file' => $file)));
+    $this->assertEqual($error['group'], $group, format_string("Group was %group", array('%group' => $group)));
+    $this->assertEqual($error['caller']['function'], $function, format_string("Function was %function", array('%function' => $function)));
+    $this->assertEqual(drupal_basename($error['caller']['file']), $file, format_string("File was %file", array('%file' => $file)));
```
Comment (quicksketch) [inline core/modules/simpletest/tests/common.test]: drupal => backdrop

File: core/modules/user/user.module [hunk 4]
```diff
@@ -2094,10 +2102,12 @@ function _user_cancel($edit, $account, $method) {
...
 
-  // After cancelling account, ensure that user is logged out.
+  // After cancelling account, ensure that user is logged out. We can't destroy
+  // their session though, as we might have information in it, and we can't
+  // regenerate it because batch API uses the session ID, we will regenerate it
+  // in _user_cancel_session_regenerate().
   if ($account->uid == $user->uid) {
-    // Destroy the current session, and reset $user to the anonymous user.
-    session_destroy();
+    $user = drupal_anonymous_user();
```
Comment (quicksketch) [inline core/modules/user/user.module]: drupal => backdrop

File: core/modules/user/user.module [hunk 5]
```diff
@@ -2105,6 +2115,17 @@ function _user_cancel($edit, $account, $method) {
...
 
 /**
+ * Finished batch processing callback for cancelling a user account.
+ *
+ * @see user_cancel()
+ */
+function _user_cancel_session_regenerate() {
+  // Regenerate the users session instead of calling session_destroy() as we
+  // want to preserve any messages that might have been set.
+  drupal_session_regenerate();
```
Comment (quicksketch) [inline core/modules/user/user.module]: drupal => backdrop

**Comment by quicksketch** [comment]
Functionality-wise this looks great but there are a couple "drupal" strings that need to be converted here. Links to http://drupal.org can stay for reference, but functions should all be renamed.

**Comment by jenlampton** [comment]
removed all instances of drupal_ in the files mentioned. pushing up to re-test.

**Pull request merged**